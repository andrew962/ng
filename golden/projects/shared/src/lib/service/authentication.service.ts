import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AuthenticationService {
  isLogged$: Observable<boolean>;
  isLoggedSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
  constructor() {
    this.isLogged$ = this.isLoggedSubject.asObservable();
  }
}
