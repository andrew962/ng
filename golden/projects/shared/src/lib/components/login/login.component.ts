import { Component, Inject, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { AuthenticationService } from '../../service/authentication.service';

@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styles: [
  ]
})
export class LoginComponent implements OnInit {
  credentials: FormGroup;
  constructor(
    @Inject('appName') public appName: string,
    fb: FormBuilder,
    private authenticationService: AuthenticationService
  ) {
    this.credentials = fb.group({
      email: [, Validators.required],
      pwd: [, Validators.required]
    })
  }

  ngOnInit(): void {
  }
  loginBtn() {
    // this.authenticationService.firebaseAuth(this.credentials.controls.email.value, this.credentials.controls.pwd.value).then(r => console.log(r))
  }
}
