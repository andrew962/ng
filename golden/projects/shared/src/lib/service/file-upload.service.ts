import { Injectable } from '@angular/core';
import { AngularFireStorage } from '@angular/fire/storage';
import { Observable, Subscription } from 'rxjs';
import { concatMap, finalize, tap } from 'rxjs/operators';
import { CurrentUser } from '../interfaces/currentUser.model';
import { AuthenticationService } from './authentication.service';
import { UserService } from './user.service';

@Injectable({
  providedIn: 'root'
})
export class FileUploadService {
  private uid: string;
  user: CurrentUser;
  authSubscription: Subscription;
  downloadURL: Observable<string>;
  constructor(
    private fStorage: AngularFireStorage,
    private auth: AuthenticationService,
    private userService: UserService
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => {
      this.user = u;
      this.uid = u?.Uid
    });
  }
  upload(file, fileName?: string, path: string = 'images') {
    let filePath: string = `${this.uid}/${path}/${fileName}`
    let fileRef = this.fStorage.ref(filePath);
    return this.fStorage.upload(filePath, file).snapshotChanges().pipe(concatMap(r => fileRef.getDownloadURL()));
  }
  getFile(filePath) {
    this.fStorage.ref(filePath).getDownloadURL().subscribe(r => {
      console.log(r)
    })
    // let filePath: string = `${this.uid}/${path}`
    // const fileRef = this.fStorage.ref(filePath);
    // this.downloadURL = fileRef.getDownloadURL();
    // return this.downloadURL.subscribe(r => {
    //   console.log(r)
    // })
  }
}
