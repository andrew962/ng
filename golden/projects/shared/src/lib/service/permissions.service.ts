import { Injectable, OnDestroy } from '@angular/core';
import { AngularFireAuth } from '@angular/fire/auth';
import { AngularFireDatabase } from '@angular/fire/database';
import { Subscription } from 'rxjs';
import { map } from 'rxjs/operators';
import { AuthenticationService } from './authentication.service';

@Injectable({
  providedIn: 'root'
})
export class PermissionsService implements OnDestroy {
  private uid: string;
  authSubscription: Subscription;
  private readonly rule = false
  constructor(
    private auth: AuthenticationService,
    private afDatabase: AngularFireDatabase,
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => this.uid = u?.Uid);
  }
  ngOnDestroy(): void {
    this.authSubscription.unsubscribe();
  }
  setPermissions() {

  }
}
