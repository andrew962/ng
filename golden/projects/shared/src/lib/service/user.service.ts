import { Injectable, OnDestroy } from '@angular/core';
import { AngularFireAuth } from '@angular/fire/auth';
import { AngularFireDatabase } from '@angular/fire/database';
import { Subscription } from 'rxjs';
import { UsersPath } from '../enums/user.enum';
import { CurrentUser } from '../interfaces/currentUser.model';
import { AuthenticationService } from './authentication.service';

@Injectable({
  providedIn: 'root'
})
export class UserService implements OnDestroy {
  private readonly rule = false
  private uid: string;
  user: CurrentUser;
  authSubscription: Subscription;
  constructor(
    private afAuth: AngularFireAuth,
    private db: AngularFireDatabase,
    private auth: AuthenticationService
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => {
      this.user = u;
      this.uid = u?.Uid
    });
  }
  ngOnDestroy(): void {
    this.authSubscription.unsubscribe();
  }

  createUser() {
    this.db.database.ref(UsersPath.DbUsersPath).child(this.uid).set({
      User: this.user,
      Permission: {
        Create: false,
        Update: false,
        Delete: false
      },
      Photos: []
    })
  }
  getUsers() {
    return this.db.database.ref(UsersPath.DbUsersPath).get()
  }
  updateUsers(data) {
    return this.db.database.ref(UsersPath.DbUsersPath).set(data);
  }
  updateUser() {
    this.db.database.ref(UsersPath.DbUsersPath).child(this.uid).update({
      Permission: {
        Create: true,
        Update: true,
        Delete: true
      }
    })
  }

}
