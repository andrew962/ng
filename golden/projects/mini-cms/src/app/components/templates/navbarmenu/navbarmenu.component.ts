import { Component, OnInit } from '@angular/core';
import { MatDialog } from '@angular/material/dialog';
import { LoginComponent } from 'projects/shared/src/lib/components/login/login.component';
import { AuthenticationService } from 'projects/shared/src/lib/service/authentication.service';
import { Observable } from 'rxjs';

@Component({
  selector: 'app-navbarmenu',
  templateUrl: './navbarmenu.component.html',
  styles: [
  ]
})
export class NavbarMenuComponent implements OnInit {
  islogged$: Observable<boolean>;
  constructor(
    private authenticationService: AuthenticationService,
    private dialog: MatDialog
  ) {
    this.islogged$ = this.authenticationService.isLogged$
  }

  ngOnInit(): void {
  }
  btnOpenDialogLogin(){
    const dialogRef = this.dialog.open(LoginComponent)
  }
}
