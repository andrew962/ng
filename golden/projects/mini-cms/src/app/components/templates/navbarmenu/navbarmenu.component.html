<nav class="navbar navbar-expand-lg navbar-light bg-light">
    <a class="navbar-brand">Mini-CMS</a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav"
        aria-controls="navbarNav" aria-expanded="false" aria-label="Home">
        <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse order-1" id="navbarNav">
        <ul class="navbar-nav">
            <li class="nav-item" [routerLinkActive]="'active'">
                <a class="nav-link" [routerLink]="'/'">Inicio</a>
            </li>
        </ul>
        <ul class="navbar-nav ml-auto">
            <li class="nav-item">
                <button class="btn btn-sm btn-primary" (click)="btnOpenDialogLogin()" [routerLink]="'login'">Iniciar
                    sesi√≥n</button>
                <!-- <a class="nav-link" [routerLink]="'login'">Login</a> -->
            </li>
        </ul>
    </div>
</nav>